### ISSUE TITLE
Crashes while lredir attempts a mounted nfs share

### ISSUE THREAD

**Issue by nnn19** [dosemu2/dosemu2#1475]
**Describe the bug**
**dosemu2** crashes while **lredir** attempts an **nfs mounted folder**.

**To Reproduce**
Create a folder **test** under home folder.
Mount an **nfs share** into **\~/test** with **auto,defaults,users,tcp,noac,nolock**
Under dosemu2
`lredir g: linux\fs/home/user/test/`
crashes the dosemu2.

And the **terminal** through which **dosemu2** is fired shows
**ERROR: MFS: failed to get xattrs, unsupported!**

**A regression?**
This bug does not exist in **dosemu**.

**Additional info**
**OS** - Fedora Workstation 33-1.2
**Arch** - x86_64.

**Comment by stsp** [comment]
Why are you not attaching the logs?
The template clearly tells you to do so.

**Comment by nnn19** [comment]
It is a different cpu. I will attach later.

**Comment by andrewbird** [comment]
> Mount an nfs share into ~/test with auto,defaults,users,tcp,noac,**nolock**

@stsp I wonder how the new OFD locking will react to that?

**Comment by stsp** [comment]
Depends on how this will work:
`err = flock(dir_fd, LOCK_EX);`
Seems like it works, or nothing
would work at all.
I wonder if that option doesn't
disable flock(), but only fcntl()
locks? Or maybe it doesn't
return an error but does nothing?
Or maybe it only means that
locks do not propagate via network,
but still work locally?
No idea...

**Comment by nnn19** [comment]
The same setup is running for years in some of my customers' sites.
OpenSUSE NFS Server
OpenSUSE Client machines with (old) dosemu runs well till now.

Need arises as the (old) dosemu could not be installed in newer Linux OS. And when **dosemu2** is tried from Fedora Client machine (as I could not successfully install dosemu2 in OpenSUSE).

Under the same Server - Client setup but with the new Fedora client with **dosemu2** the nfs share, when lredir-ed crashes the dosemu instance.

**Comment by stsp** [comment]
No logs - no talks. :)
And don't forget to update to latest first.

**Comment by nnn19** [comment]
Yes. Understand. I will reach that CPU and give full feedback soon.

**Comment by stsp** [comment]
We will have the opensuse builds
eventually. But so far I am fighting
with a very painful problems on that
front:
https://github.com/jwt27/build-gcc/issues/22
https://github.com/jwt27/build-gcc/issues/21
Fortunately, with hints from a djgpp
guru I am getting some progress.
That besides fighting with the OBS
itself - the most difficult to grok
build system I've ever seen.

**Comment by nnn19** [comment]
For sure - not pushing for OpenSUSE build now.
I am an extensive user of (old) dosemu for 2+ decades.
Will help in what ever possible ways I can.

**Comment by nnn19** [comment]
`zypper in build-gcc-1620588549.ff7f43d-9.1.x86_64.rpm `
Problem: nothing provides libc.so.6(GLIBC_2.33)(64bit) needed by build-gcc-1620588549.ff7f43d-9.1.x86_64

I tried installing
`zypper in https://ftp.lysator.liu.se/pub/opensuse/tumbleweed/repo/oss/x86_64/glibc-2.33-5.2.x86_64.rpm`
which gave a very huge deinstallation list - hence dropped
and

`zypper in dosemu2-1620602133.72f72c6b2-8.1.x86_64.rpm `
Problem: nothing provides libSDL2-2.0.so.0(SUSE_2.0.10)(64bit) needed by dosemu2-1620602133.72f72c6b2-8.1.x86_64

OS: OpenSUSE Leap 15.2
glibc: 2.26-lp152.26.6.1.x86_64
Is this the reason? or any other source to get glibc 2.33?

**Comment by stsp** [comment]
OK so if you have Leap, this
will have to wait. I wasn't able
to get djgpp working on leap
because leap doesn't support
/usr/libexec location.

@mateuszviste what distro do
you have? Can you test the
tumbleweed repo?

**Comment by nnn19** [comment]
The Tumbleweed 20210507 iso download is going on.

Will update as soon as I...

**Comment by mateuszviste** [comment]
I am running Leap 15.2. It is very nice that DOSEMU2 is on the OpenSUSE software list now - could you please enable also the builds for Leap distros? (this should be as simple as checking a few extra checkboxes in the "repositories" tab on build.opensuse.org)

**Comment by nnn19** [comment]
As a matter of interest...
The Tumbleweed 20210507 iso download is going on.
Will update as soon as I...

**Comment by stsp** [comment]
Hmm, I can think of a libexec fix perhaps...

**Comment by nnn19** [comment]
OpenSUSE Tumbleweed 20210507
build- ok
comcom32- ok
fdpp- ok
dosemu2- generates error.
Problem: nothing provides systemd-udev >= 234 needed by dosemu2-1620602133.72f72c6b2-8.1.x86_64

**Comment by stsp** [comment]
Should now be fixed, please re-try.

**Comment by nnn19** [comment]
Remember - this is not an urgent task.
Anyone can wait for a clear installation under **Leap** than under **Tumbleweed**.
And for sure ratio of L:T users will be 90% : 10%

**Comment by stsp** [comment]
Should now be fixed.
The new build started automatically,
so please update in 10 minutes.

**Comment by stsp** [comment]
You can yourself open any
ticket and copy the text there.
I don't see a big need deleting
the existing comments from
this thread.

Please attach the strace log,
like this:
`strace -f -o ~/strace.log dosemu.bin`
This will start dosemu, you
reproduce the problem, then
attach the ~/strace.log

**Comment by nnn19** [comment]
The sttrace
[strace.log](https://github.com/dosemu2/dosemu2/files/6453275/strace.log)

And Yes. It is permission issue - my oversight.

Solved.
